musicbot/bot.py

```diff
@@ -3129,6 +3132,41 @@ def _get_url() -> str:
                 )
             return url
 
+        if option in ["+all", "add_all"]:
```

Comment:
This will work, but as you mentioned wanting to use `!autoplaylist [+|add] all` in discord, for that you would want to inspect the value of `opt_url` directly.  If no "url" command parameter is given `opt_url` is simply an empty string as per its default value in the function definition.   

Since this command currently does not allow users to add songs which are not URLs, reserving the word "all" wouldn't be an issue.  If later we decide song titles should be allowed through this function, this command would need some refactor to allow that anyways. 

So dealers choice here, I'm good with this if you are.

Reply:
I've taken the route of reserving "all" as a special case of opt_url.